Utilities/WDLs/DownsampleAndCollectCoverage.wdl

```diff
@@ -0,0 +1,172 @@
+            coverage_intervals = coverage_intervals,
+            ref_fasta = ref_fasta,
+            ref_fasta_index = ref_fasta_index,
+            read_length = read_length,
+            use_fast_algorithm = use_fast_algorithm,
+            docker = docker,
+            picard_jar_override = picard_jar_override,
+            preemptible = preemptible
+    }
+
+    output {
+        File downsampled_cram = Downsample.downsampled_cram
+        File downsampled_cram_index = Downsample.downsampled_cram_index
+        Float downsampled_mean_coverage = CollectDownsampledCoverage.mean_coverage
+        Float? original_mean_coverage = CollectOriginalCoverage.mean_coverage
+    }
+}
+
+task CollectWgsMetrics {
+    input {
+        File input_cram
+        File input_cram_index
+        File? coverage_intervals
+        File ref_fasta
+        File ref_fasta_index
+        Int read_length
+        Boolean use_fast_algorithm
+
+        String docker
+        File? picard_jar_override
+
+        Int preemptible
+    }
+
+    Int disk_size = ceil(size(input_cram, "GiB") + size(ref_fasta, "GiB") + size(ref_fasta_index, "GiB")) + 50
+
+    String output_basename = sub(sub(basename(input_cram), "\\.bam$", ""), "\\.cram$", "")
+
+    command <<<
+        set -e -o pipefail
+
+        ~{if defined(picard_jar_override) then "java -Xms2000m -Xmx2500m -jar " + picard_jar_override else 'gatk --java-options "-Xms2000m -Xmx2500m"'} \
+        CollectWgsMetrics \
+        -INPUT ~{input_cram} \
+        -VALIDATION_STRINGENCY SILENT \
+        -REFERENCE_SEQUENCE ~{ref_fasta} \
+        ~{"-INTERVALS " + coverage_intervals} \
+        -OUTPUT ~{output_basename}.wgs_metrics \
+        -READ_LENGTH ~{read_length} \
+        -USE_FAST_ALGORITHM ~{use_fast_algorithm}
+
+        cat ~{output_basename}.wgs_metrics | grep -v -e "^#" -e "^$" -e "^GENOME_TERRITORY" | head -n 1 | awk '{ print $2 }' > ~{output_basename}.mean_coverage
+    >>>
+    runtime {
+        docker: docker
+        preemptible: preemptible
+        memory: "3000 MiB"
+        disks: "local-disk " + disk_size + " HDD"
+    }
+    output {
+        Float mean_coverage = read_float("~{output_basename}.mean_coverage")
+    }
+}
+
+task Downsample {
+    input{
+        File input_cram
+
+        Float? downsample_probability
+        Float? target_coverage
+        Float? original_coverage
+
+        String downsample_strategy
+        File ref_fasta
+        File ref_fasta_index
+
+        String docker
+        File? picard_jar_override
+        Int preemptible
+        Int additional_disk_gb = 200
+    }
+    Int disk_size = ceil(3.5 * size(input_cram, "GiB") + 20 + additional_disk_gb)
+
+    String output_basename = sub(sub(basename(input_cram), "\\.bam$", ""), "\\.cram$", "")
+
+
+    command {
+        set -e -o pipefail
+
+        PROBABILITY=~{if defined(downsample_probability) then downsample_probability else "$(bc -l <<< 'scale=2; " + target_coverage + "/" + original_coverage + "')"}
```

Comment:
You might want to handle the edge case if `target_coverage` is accidentally larger than `original_coverage` to set this to 1.